qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -179,21 +178,18 @@ def _wire_symbol_from_gate(gate: cirq.Gate, signature: Signature, soq: 'Soquet')
     return directional_text_box(text=symbol, side=soq.reg.side)
 
 
-def _add_my_tensors_from_gate(
-    gate: cirq.Gate,
-    signature: Signature,
-    short_name: str,
-    tn: qtn.TensorNetwork,
-    tag: Any,
-    *,
-    incoming: Dict[str, 'SoquetT'],
-    outgoing: Dict[str, 'SoquetT'],
-):
-    if not cirq.has_unitary(gate):
-        raise NotImplementedError(
-            f"CirqGateAsBloq.add_my_tensors is currently supported only for unitary gates. "
-            f"Found {gate}."
-        )
+def _tensor_shape_from_signature(signature: Signature) -> Tuple[int, ...]:
+    return tuple(
+        [
+            *itertools.chain.from_iterable(
+                (2**reg.bitsize,) * int(np.prod(reg.shape))
+                for reg in [*signature.rights(), *signature.lefts()]
+            )
+        ]
+    )
+
+
+def _tensor_data_from_unitary_and_signature(inp_unitary: np.ndarray, signature: Signature):
```

Comment:
move to a module in `qualtran.simulation.tensor`?

Reply:
done